charts/azure-managed-cluster/README.md

```diff
@@ -0,0 +1,154 @@
+```bash
+export AZURE_CLIENT_ID="" (if using Service Principal)
+
+export AZURE_CLIENT_SECRET="" (if using Service Principal)
+
+export AZURE_SUBSCRIPTION_ID=""
+
+export AZURE_TENANT_ID=""
+
+```
+
+Load Environment variables
+```bash
+source clusterctl.env
+```
+
+Create a KIND cluster(Kind clusters only work service principal)):
+
+```bash
+kind create cluster --name capi-helm
+```
+
+
+Initialize Cluster API and install Azure CAPZ provider version 
+
+```bash
+clusterctl init --infrastructure azure
+```
+
+Deploy a cluster with Helm (please customize parameters as required)
+
+**Using Service Principal:**
+
+```bash
+helm install capz1 charts/azure-managed-cluster/  \
+--namespace default \
+--set subscriptionID="${AZURE_SUBSCRIPTION_ID}" \
+--set identity.clientId="${AZURE_CLIENT_ID}" \
+--set identity.clientSecret="${AZURE_CLIENT_SECRET}" \
+--set identity.type=ServicePrincipal \
+--set identity.tenantId="${AZURE_TENANT_ID}" \
+--set cluster.resourceGroupName=aksclusters \
+--set cluster.nodeResourceGroupName=capz1 \
+--set cluster.name=aks1 \
+--set agentpools.agentpool0.name=capz1np0 \
+--set agentpools.agentpool0.nodecount=1 \
+--set agentpools.agentpool0.sku=Standard_B4ms \
+--set agentpools.agentpool0.osDiskSizeGB=100 \
+--set agentpools.agentpool0.mode=System \
+--set agentpools.agentpool1.name=capz1np1 \
+--set agentpools.agentpool1.nodecount=1 \
+--set agentpools.agentpool1.sku=Standard_B4ms \
+--set agentpools.agentpool1.osDiskSizeGB=10 \
+--set agentpools.agentpool1.mode=User 
+```
+
+or more simply (after you edit the values file with your own values):
+
+```bash
+helm install capz1 charts/azure-managed-cluster/ --values values.yaml \
+--namespace default \
+--set controlplane.sshPublicKey="$(cat ~/.ssh/id_rsa.pub)" \
+--set subscriptionID="${AZURE_SUBSCRIPTION_ID}" \
+--set identity.clientID="${AZURE_CLIENT_ID}" \
+--set identity.tenantID="${AZURE_TENANT_ID}" \
+--set identity.clientSecret="${AZURE_CLIENT_SECRET}" \
+--set identity.type=ServicePrincipal 
+```
+
+**Using Managed Identity**
+
+NB: Ensure the AZURE_USER_ASSIGNED_IDENTITY_RESOURCE_ID is set by getting the resource id of the managed identity created in Azure
+
+
+```bash
+
+helm install capz1 charts/azure-managed-cluster/  \
+--namespace default \
+--set subscriptionID="${AZURE_SUBSCRIPTION_ID}" \
+--set identity.clientID="${AZURE_CLIENT_ID}" \
+--set identity.tenantID="${AZURE_TENANT_ID}" \
+--set identity.type=UserAssignedMSI \
+--set identity.resourceID="${AZURE_USER_ASSIGNED_IDENTITY_RESOURCE_ID}" 
+
+
+
+Check the status with:
+```
+kubectl get cluster-api
+kubectl  logs -n capz-system -l control-plane=capz-controller-manager -c manager -f
+```
+
+Get the credentials
+
+```
+kubectl get secret capi-helm-kubeconfig -o yaml -o jsonpath={.data.value} | base64 --decode > aks1.kubeconfig
+```
+
+Test the cluster!
+
+```
+kubectl --kubeconfig=aks1.kubeconfig cluster-info
+```
+
```

Comment:
Remove extra newlines

Reply:
I'm hoping this is okay now since i've made updates to this